content/asm_6.md

```diff
@@ -1,181 +1,561 @@
+- `subsd` - Subtracts a double-precision floating-point values.
+- `mulss` - Multiplies a single-precision floating-point values.
+- `mulsd` - Multiplies a double-precision floating-point values.
+- `divss` - Divides a single-precision floating-point values.
+- `divsd` - Divides a double-precision floating-point values.
+
+All of these instructions expect two operands to execute the given operation of addition, subtraction, multiplication, or division. After the operation is executed, the result is stored in the first operand.
+
+### Floating-Point control instructions
+
+As we already know, the main goal of these instructions is to manage the control flow of our programs. For integer comparisons, we used the `cmp` instruction. But this instruction does not work with floating-point values, even though the result of the comparison is controlled by the same `rflags` registers that we saw in the [Data manipulation](./asm_4.md#control-transfer-instructions) chapter. The general instruction to compare floating-point numbers is `cmpss`. Instead of setting the value of the flag based on the result comparison, the instruction stores the result of the comparison in the destination register.
+
+For example:
 
 ```assembly
-mov destination, source
+;; Compare xmm0 < xmm1 and store the result in the xmm0
+cmpss xmm0, xmm1, 0x1
 ```
 
-With GNU assembler we have back order i.e.:
+The third operand of the instruction, identifies the operator:
+
+| 3rd operand value | Meaning                                                                |
+|-------------------|------------------------------------------------------------------------|
+|                 0 | ==                                                                     |
+|                 1 | <                                                                      |
+|                 2 | <=                                                                     |
+|                 3 | Check that one of operands is [Nan](https://en.wikipedia.org/wiki/NaN) |
+|                 4 | !=                                                                     |
+|                 5 | >=                                                                     |
+|                 6 | >                                                                      |
+|                 7 | Checks that both operands are [Nan](https://en.wikipedia.org/wiki/NaN) |
+
+## Example
+
+After we got familiar with the basics of the new topic, it's time to write some code. This time, we will try to write a program which reads the user input, builds two [vectors](https://en.wikipedia.org/wiki/Vector_(mathematics_and_physics)) of floating-point numbers based on it, and calculates the [dot product](https://en.wikipedia.org/wiki/Dot_product) of these two vectors. This operation is widely used in machine learning. It will be interesting to try to implement it using the assembly programming language.
```

Comment:
```suggestion
After we got familiar with the basics of the new topic, it's time to write some code. This time, we will try to write a program which reads the user input, builds two [vectors](https://en.wikipedia.org/wiki/Vector_(mathematics_and_physics)) of floating-point numbers based on it, and calculates the [dot product](https://en.wikipedia.org/wiki/Dot_product) of these two vectors. This operation is widely used in machine learning. It will be interesting to implement it using the assembly programming language.
```